test/workbox-core/sw/_private/test-cacheWrapper.mjs

```diff
@@ -221,6 +221,83 @@ describe(`cacheWrapper`, function() {
       }))).to.be.true;
     });
 
+    it(`should call cacheKeyWillBeUsed`, async function() {
+      const cacheName = 'cacheKeyWillBeUsed-test-cache';
+      const cache = await caches.open(cacheName);
+      sandbox.stub(self.caches, 'open').resolves(cache);
+      const cachePutStub = sandbox.stub(cache, 'put').resolves();
+
+      const firstPluginReturnValue = new Request('/firstPlugin');
+      const firstPlugin = {
+        cacheKeyWillBeUsed: () => firstPluginReturnValue,
+      };
+
+      const secondPlugin = {
+        // This string will be converted to a Request.
+        cacheKeyWillBeUsed: () => '/secondPlugin',
+      };
+
+      const spyOne = sandbox.spy(firstPlugin, 'cacheKeyWillBeUsed');
+      const spyTwo = sandbox.spy(secondPlugin, 'cacheKeyWillBeUsed');
+
+      const initialRequest = new Request('/noPlugin');
+      const response = new Response('Test response.');
+
+      await cacheWrapper.put({
+        response,
+        cacheName,
+        request: initialRequest,
+        plugins: [
+          firstPlugin,
+          {}, // Intentionally empty to ensure it's filtered out.
+          secondPlugin,
+        ],
+      });
+
+      expect(spyOne.calledOnceWith({
+        mode: 'write',
+        request: initialRequest,
+      })).to.be.true;
+      expect(spyOne.thisValues[0]).to.eql(firstPlugin);
+
+      expect(spyTwo.calledOnceWith({
+        mode: 'write',
+        request: firstPluginReturnValue,
+      })).to.be.true;
+      expect(spyTwo.thisValues[0]).to.eql(secondPlugin);
+
+      expect(cachePutStub.calledOnce).to.be.true;
+      // Check the url of the Request passed to cache.put().
+      expect(cachePutStub.args[0][0].url).to.eql(`${self.location.origin}/secondPlugin`);
+      expect(cachePutStub.args[0][1]).to.eql(response);
+    });
+
+    it(`should throw an error in development when the cacheKeyWillBeUsed return is invalid `, async function() {
+      if (process.env.NODE_ENV === 'production') this.skip();
+
+      const request = new Request('/');
+      const response = new Response('');
+
+      const plugin = {
+        cacheKeyWillBeUsed: () => null,
+      };
+      const spy = sandbox.spy(plugin, 'cacheKeyWillBeUsed');
+
+      await expectError(async () => {
+        await cacheWrapper.put({
+          cacheName: 'test-cache-name',
+          request,
+          response,
+          plugins: [plugin],
+        });
+      }, 'incorrect-class');
+
+      expect(spy.calledOnceWith({
+        request,
+        mode: 'write',
+      }));
```

Comment:
Add `.to.be.true`, otherwise this won't check anything.

Reply:
Ugh.